### PR TITLE
Modernize some C++

### PR DESCRIPTION
**Pull request by ThomasHabets**

### CODE REVIEW

**Comment by mbr0wn** [comment]
Our min gcc on master is 4.8.4 right now.

**Comment by mbr0wn** [comment]
OK CLA is a-OK it seems.

File: gr-blocks/lib/pack_k_bits_bb_impl.h [hunk 1]
```diff
@@ -32,11 +32,10 @@ namespace blocks {
 class pack_k_bits_bb_impl : public pack_k_bits_bb
 {
 private:
-    kernel::pack_k_bits* d_pack;
+    const kernel::pack_k_bits d_pack;
```
Comment (marcusmueller) [inline gr-blocks/lib/pack_k_bits_bb_impl.h]: I like! having a private variable be a locally allocated and deallocated and pointed to object makes no sense. Use C++'s own lifetime management :1

Comment (marcusmueller) [reply gr-blocks/lib/pack_k_bits_bb_impl.h]: :100:

File: gr-digital/lib/fll_band_edge_cc_impl.cc [hunk 2]
```diff
@@ -180,23 +174,21 @@ void fll_band_edge_cc_impl::design_filter(float samps_per_sym,
...
 
 void fll_band_edge_cc_impl::print_taps()
 {
-    unsigned int i;
-
     printf("Upper Band-edge: [");
-    for (i = 0; i < d_taps_upper.size(); i++) {
-        printf(" %.4e + %.4ej,", d_taps_upper[i].real(), d_taps_upper[i].imag());
+    for (const auto& tap : d_taps_upper) {
+        printf(" %.4e + %.4ej,", tap.real(), tap.imag());
```
Comment (marcusmueller) [inline gr-digital/lib/fll_band_edge_cc_impl.cc]: one day I'll have to remember to replace all printf with logger calls.

File: gr-digital/lib/symbol_sync_cc_impl.cc [hunk 3]
```diff
@@ -105,14 +97,14 @@ symbol_sync_cc_impl::symbol_sync_cc_impl(enum ted_type detector_type,
         throw std::out_of_range("output samples per symbol must be > 0");
 
     // Timing Error Detector
-    d_ted = timing_error_detector::make(detector_type, slicer);
-    if (d_ted == NULL)
+    d_ted.reset(timing_error_detector::make(detector_type, slicer));
+    if (d_ted == nullptr)
```
Comment (marcusmueller) [inline gr-digital/lib/symbol_sync_cc_impl.cc]: {}

Comment (marcusmueller) [reply gr-digital/lib/symbol_sync_cc_impl.cc]: also, un-idiomatic: `d_ted` is a `unique_ptr`, and as such has `operator bool` that does exactly this, so just `if(!d_ted)` would be the right thing.

File: gr-digital/lib/symbol_sync_cc_impl.cc [hunk 4]
```diff
@@ -105,14 +97,14 @@ symbol_sync_cc_impl::symbol_sync_cc_impl(enum ted_type detector_type,
...
         throw std::runtime_error("unable to create timing_error_detector");
 
     // Interpolating Resampler
-    d_interp = interpolating_resampler_ccf::make(
-        interp_type, d_ted->needs_derivative(), n_filters, taps);
-    if (d_interp == NULL)
+    d_interp.reset(interpolating_resampler_ccf::make(
+        interp_type, d_ted->needs_derivative(), n_filters, taps));
+    if (d_interp == nullptr)
```
Comment (marcusmueller) [inline gr-digital/lib/symbol_sync_cc_impl.cc]: same as above

**Pull request merged**